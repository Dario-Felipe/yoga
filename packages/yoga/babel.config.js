module.exports = {
  extends: '../../babel.config.js',
  plugins: ['transform-import-css'],
};
