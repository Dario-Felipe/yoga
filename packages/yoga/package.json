{
  "name": "@gympass/yoga",
  "version": "5.2.0-alpha.4",
  "description": "Gympass component library",
  "main": "./src",
  "sideEffects": false,
  "scripts": {
    "build": "yarn build:web && yarn build:native && yarn build:esm",
    "build:web": "NODE_ENV=web babel ./src --out-dir dist/cjs",
    "build:esm": "NODE_ENV=esm babel ./src --out-dir dist/esm",
    "build:native": "NODE_ENV=native babel ./src --out-dir dist/cjs",
    "prebuild": "rm -rf ./dist",
    "postbuild": "cp -r ./src/Fonts ./dist/fonts",
    "prepublishOnly": "node ../../scripts/prepublish.js --rn"
  },
  "keywords": [
    "Gympass",
    "components",
    "styled-components",
    "react",
    "design-system"
  ],
  "homepage": "https://github.com/Gympass/yoga#readme",
  "license": "MIT",
  "publishConfig": {
    "access": "public",
    "directory": "dist"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Gympass/yoga.git"
  },
  "bugs": {
    "url": "https://github.com/Gympass/yoga/issues"
  },
  "dependencies": {
    "@gympass/yoga-common": "^0.1.2-alpha.4",
    "@gympass/yoga-helpers": "^0.1.1-alpha.0",
    "@gympass/yoga-icons": "^0.11.2-alpha.5",
    "@gympass/yoga-tokens": "^1.1.0-alpha.0",
    "@ptomasroos/react-native-multi-slider": "^2.2.2",
    "downshift": "^6.1.0",
    "prop-types": "^15.7.2",
    "proxy-polyfill": "^0.3.2",
    "rc-slider": "^9.7.1",
    "react-google-font-loader": "^1.1.0"
  },
  "devDependencies": {
    "@testing-library/react": "^11.2.5",
    "@testing-library/react-native": "^7.2.0",
    "babel-plugin-inline-react-svg": "^2.0.1",
    "styled-components": "^5.2.1"
  },
  "peerDependencies": {
    "styled-components": "^5.2.1"
  }
}
