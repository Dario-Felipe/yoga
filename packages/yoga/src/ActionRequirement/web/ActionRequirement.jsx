import React from 'react';
import { arrayOf, node, oneOfType, string } from 'prop-types';
import styled from 'styled-components';
import { media } from '@gympass/yoga-helpers';
import {
  Actions,
  Title,
  PrimaryButton,
  SecondaryButton,
} from './ActionRequirementStyles';
import Text from '../../Text';
import Box from '../../Box';

const StyledActionRequirement = styled.div`
  display: flex;
  height: 100%;
  ${media.xxs`
    flex-direction: column;
  `}
  ${media.lg`
    flex-direction: row-reverse;
  `}
`;

const Content = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;

  ${media.xxs`
    width: auto;
    margin-right: 0;
  `}

  ${media.lg`
    width: 520px;
    margin-right: 120px;
  `}
`;

const BoxIllustration = styled(Box)`
  flex-grow: 1;
  display: flex;
  justify-content: center;
  align-items: center;
`;

function isChildFromComponent(child, component) {
  return child.type.displayName === component.displayName;
}

function ActionRequirement(props) {
  const { title, description, children, checkable, illustration, list } = props;

  let primaryButton;
  let secondaryButton;

  function defineCompoundComponents() {
    React.Children.forEach(children, child => {
      if (isChildFromComponent(child, PrimaryButton)) primaryButton = child;
      if (isChildFromComponent(child, SecondaryButton)) secondaryButton = child;
    });
  }

  defineCompoundComponents();
  return (
    <StyledActionRequirement {...props}>
      {illustration && <BoxIllustration>{illustration}</BoxIllustration>}

      <Content>
        <Title>{title}</Title>
        <Text mt="small" color="deep">
          {description}
        </Text>
        {list && (
          <Text mt="large" color="deep">
            <Box>{list}</Box>
          </Text>
        )}
        {checkable && <Box mt="xxlarge">{checkable}</Box>}
        <Actions mt="xlarge">
          {primaryButton}
          {secondaryButton}
        </Actions>
      </Content>
    </StyledActionRequirement>
  );
}

ActionRequirement.propTypes = {
  title: string.isRequired,
  children: oneOfType([arrayOf(node), node]),
  description: string.isRequired,
  checkable: oneOfType([arrayOf(node), node]),
  illustration: oneOfType([arrayOf(node), node]),
  list: oneOfType([arrayOf(node), node]),
};

ActionRequirement.defaultProps = {
  children: undefined,
  checkable: undefined,
  illustration: undefined,
  list: undefined,
};

export default ActionRequirement;
