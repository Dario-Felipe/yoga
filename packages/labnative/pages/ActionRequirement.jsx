import React from 'react';
import { View } from 'react-native';
import { ActionRequirement, Box, Checkbox, Text } from '@gympass/yoga';
import { Img2d01 } from '@gympass/yoga-illustrations';
import { Dumbbell, User, Star } from '@gympass/yoga-icons';

function handleGetCheckableContent() {
  return (
    <Box display="flex" flexDirection="row" alignItems="center">
      <Checkbox />
      <Text>I have read and agree to the Terms of Use and Privacy Policy.</Text>
    </Box>
  );
}

function handleGetIllustration() {
  return <Img2d01 width={375} height={375} />;
}

function handleGetList() {
  return (
    <>
      <Text>
        <Dumbbell /> 1x/day standard access
      </Text>
      <Text mt="small">
        <User /> 4x/month 1-on-1 sessions
      </Text>
      <Text mt="small">
        <Star /> 2x/month premium classes
      </Text>
    </>
  );
}

function ActionRequirementPage() {
  return (
    <View>
      <ActionRequirement
        title="Welcome to the world of feeling good!"
        description="Letâ€™s make it a good day!"
        illustration={handleGetIllustration()}
        checkable={handleGetCheckableContent()}
        list={handleGetList()}
      />
    </View>
  );
}

export default ActionRequirementPage;
