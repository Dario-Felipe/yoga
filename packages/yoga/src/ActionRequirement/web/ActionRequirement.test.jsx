import React from 'react';
import { render } from '@testing-library/react';
import { Text } from '@gympass/yoga';
import { Dumbbell, User, Star } from '@gympass/yoga-icons';
import ActionRequirement from './ActionRequirement';
import { ThemeProvider } from '../..';

function handleGetList() {
  return (
    <>
      <Text>
        <Dumbbell /> 1x/day standard access
      </Text>
      <Text mt="small">
        <User /> 4x/month 1-on-1 sessions
      </Text>
      <Text mt="small">
        <Star /> 2x/month premium classes
      </Text>
    </>
  );
}

describe('<ActionRequirement />', () => {
  it('should default match snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <ActionRequirement />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with list', () => {
    const { container } = render(
      <ThemeProvider>
        <ActionRequirement list={handleGetList()} />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
