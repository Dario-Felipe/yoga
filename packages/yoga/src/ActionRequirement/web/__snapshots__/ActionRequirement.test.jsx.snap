// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionRequirement /> should match snapshot 1`] = `
.c3 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;
  font-family: Rubik;
  color: #231B22;
  margin-top: 16px;
  color: #6B6B78;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;
  font-family: Rubik;
  color: #231B22;
  margin-top: 24px;
  color: #6B6B78;
}

.c2 {
  margin: 0;
  padding: 0;
  font-size: 24px;
  line-height: 32px;
  font-weight: 500;
  font-family: Rubik;
  color: #231B22;
  margin-bottom: 6px;
}

.c5 {
  margin-top: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:769px) {
  .c2 {
    max-width: 700px;
    font-size: 48px;
    line-height: 48px;
  }
}

@media (min-width:769px) {
  .c5 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    margin-top: 32px;
  }
}

@media (min-width:0px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:1024px) {
  .c0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:0px) {
  .c1 {
    width: auto;
    margin-right: 0;
  }
}

@media (min-width:1024px) {
  .c1 {
    width: 520px;
    margin-right: 120px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <p
        class="c2"
      />
      <p
        class="c3"
        color="deep"
      />
      <p
        class="c4"
        color="deep"
      />
      <div
        class="c5"
      />
    </div>
  </div>
</div>
`;
